src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/aws/AWSEC2JCloudsProvider.java

```diff
@@ -258,6 +260,8 @@ private void addOptions(Template template, Options options, Infrastructure infra
             // Have we defined an explicit list of ports to be opened ?
             int[] ports = options.getPortsToOpen();
             String securityGroupName = getAutoGeneratedSecurityGroupName(infrastructure.getId());
+            Set<Map.Entry<String, String>> infraSg = autoGeneratedSecurityGroups.getOrDefault(infrastructure.getId(),
+                                                                                              new HashSet<>());
```

Comment:
Set of Map.Enty, that's a very strange structure ? I will need some clarification

Reply:
You're fully right. That's a Map!!! I really don't know why I complicated the struct so much lol.

Reply:
You're fully right. That's a Map!!! I really don't know why I complicated the struct so much lol.